modules/base/errors/module.py

```diff
@@ -49,308 +51,662 @@ async def on_command_error(
+            return (
+                _(ctx, "Check failure"),
+                _(ctx, "I lack the {perms} permission").format(perms=perms),
+                True,
+            )
+        # Exception raised when the command invoker lacks a role to run a command.
+        elif isinstance(error, commands.MissingRole):
+            return (
+                _(ctx, "Check failure"),
+                _(ctx, "You have to have a {role} role for that").format(
+                    role=f"`{error.missing_role!r}`",
                 ),
                 True,
             )
-
-        # non-critical nextcord exceptions
-        if type(error) == nextcord.NoMoreItems or isinstance(
-            error, nextcord.HTTPException
-        ):
-            error_message = {
-                "NoMoreItems": _(ctx, "pumpkin.py exception"),
-                "HTTPException": _(ctx, "An environment variable is missing"),
-            }
+        # Exception raised when the bot's member lacks a role to run a command.
+        elif isinstance(error, commands.BotMissingRole):
             return (
-                _(ctx, "Error"),
-                error_message.get(
-                    type(error).__name__, _(ctx, "An unexpected error occurred")
+                _(ctx, "Check failure"),
+                _(ctx, "I lack the role {role}").format(
+                    role=f"`{error.missing_role!r}`"
                 ),
                 True,
             )
+        # Exception raised when the command invoker lacks any of the roles specified to run a command.
+        elif isinstance(error, commands.MissingAnyRole):
+            roles = ", ".join(f"**{r!r}**" for r in error.missing_roles)
+            return (
+                _(ctx, "Check failure"),
+                _(ctx, "You have to have one role of {roles}").format(roles=roles),
+                True,
+            )
+        # Exception raised when the bot's member lacks any of the roles specified to run a command.
+        elif isinstance(error, commands.BotMissingAnyRole):
+            roles = ", ".join(f"**{r!r}**" for r in error.missing_roles)
+            return (
+                _(ctx, "Check failure"),
+                _(ctx, "I need one of roles {roles}").format(roles=roles),
+                True,
+            )
+        # Exception raised when a channel does not have the required NSFW setting.
+        elif isinstance(error, commands.NSFWChannelRequired):
+            return (
+                _(ctx, "Check failure"),
+                _(ctx, "This command can be used only in NSFW channels."),
+                True,
+            )
+        # Just in case we missed something
+        else:
+            return (
+                _(ctx, "Nextcord library error"),
+                _(ctx, "Check failure"),
+                False,
+            )
 
-        # critical nextcord exceptions
-        if isinstance(error, nextcord.DiscordException):
+    async def handle_UserInputError(ctx, error) -> Tuple[str, str, bool]:
```

Comment:
This should have `@staticmethod` decorator

Reply:
also fixed in the force push